### PR TITLE
start the next serial build immediately after a build is canceled

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

**Comment by bparees** [comment]
[testextended][extended:core(builds)]

**Comment by bparees** [comment]
@csrwng ptal

**Comment by bparees** [comment]
[test]

File: pkg/build/controller/controller.go [hunk 1]
```diff
@@ -273,6 +276,19 @@ func (bc *BuildController) resolveOutputDockerImageReference(build *buildapi.Bui
 	return ref, nil
 }
 
+func handleBuildCompletion(build *buildapi.Build, runPolicies []policy.RunPolicy) {
+	runPolicy := policy.ForBuild(build, runPolicies)
```
Comment (csrwng) [inline pkg/build/controller/controller.go]: looking at this code again, should we not determine if the build is complete first before getting the run policy and calling onComplete on it?

**Comment by csrwng** [comment]
Just one question, otherwise LGTM

File: pkg/build/controller/controller.go (hunk 1 above)
Comment (bparees) [inline pkg/build/controller/controller.go]: yeah that does seem like a logical optimization.

**Comment by bparees** [comment]
@csrwng updated

**Comment by csrwng** [comment]
LGTM

**Comment by bparees** [comment]
[merge]

**Comment by bparees** [comment]
[merge]

**Pull request merged**